### PR TITLE
Feature 104 finalize replies screen

### LINKED ISSUE #104: List replies of specific form
As a user I'd like to see list of replies of chosen form

### PR DESCRIPTION
**Pull request by EVP1989**
Resolve #104

### CODE REVIEW

File: src/types/index.ts [hunk 1]
```diff
@@ -91,12 +91,17 @@ export interface Dictionary<T> {
 }
 
 export enum ApplicationRoles {
```
Comment (anttileppa) [inline src/types/index.ts:93]: Missing tsdocs

File: src/types/index.ts [hunk 2]
```diff
@@ -91,12 +91,17 @@ export interface Dictionary<T> {
...
-  ADMIN = "admin",
-  MANAGER = "manager",
-  USER = "user",
+  METAFORM_ADMIN = "metaform-admin",
+  METAFORM_MANAGER = "metaform-manager",
+  METAFORM_USER = "metaform-user",
   ANONYMOUS = "anonymous"
 }
 
+export enum ReplyStatus {
```
Comment (anttileppa) [inline src/types/index.ts:100]: Missing tsdocs

File: src/types/index.ts [hunk 3]
```diff
@@ -91,12 +91,17 @@ export interface Dictionary<T> {
...
+  WAITING = "Odottaa",
```
Comment (anttileppa) [inline src/types/index.ts:101]: Why this is hard coded in Finnish?

File: src/components/screens/forms/form-replies-screen.tsx [hunk 4]
```diff
@@ -0,0 +1,278 @@
...
+
+  const [ rows, setRows ] = useState<any[]>([]);
+  const [ loading, setLoading ] = useState(false);
+  const [ managementListColumns, setManagementListColumns ] = useState<MetaformField[]>();
+  const [ metaform, setMetaform ] = useState<Metaform>();
+  const [ deletableReplyId, setDeletableReplyId ] = useState<string | undefined>(undefined);
+  const [ showAllReplies, setShowAllReplies ] = useState(false);
+
+  const useparams = useParams();
+  const { formId } = useparams;
```
Comment (anttileppa) [inline src/components/screens/forms/form-replies-screen.tsx:38]: Test whether formId is null here

File: src/components/screens/forms/form-replies-screen.tsx [hunk 5]
```diff
@@ -0,0 +1,278 @@
...
+    } catch (e) {
+      errorContext.setError(strings.errorHandling.adminRepliesScreen.fetchReplies, e);
+    }
+  
+    setLoading(false);
+  };
+
+  useEffect(() => {
+    setup();
+  }, [showAllReplies]);
```
Comment (anttileppa) [inline src/components/screens/forms/form-replies-screen.tsx:221]: Is it really necessary to reaload everything when showAllReplies changes? Couldn't we instead use filtered version of rows?

File: src/components/screens/forms/form-replies-screen.tsx [hunk 6]
```diff
@@ -0,0 +1,278 @@
...
+        <NavigationTab
+          text={ strings.navigationHeader.formsScreens.formDataScreen }
+          to=":formId/history"
+        />
+      </NavigationTabContainer>
+      <DataGrid
+        loading={ loading }
+        rows={ rows ?? [] }
+        columns={ getGridColumns() ?? [] }
+        getRowId={ row => (row.created) }
```
Comment (anttileppa) [inline src/components/screens/forms/form-replies-screen.tsx:268]: Created is not unique row id, use id instead

File: src/components/screens/forms/form-replies-screen.tsx [hunk 7]
```diff
@@ -0,0 +1,278 @@
...
+
+  /**
+   * Builds a row for the table
+   * 
+   * @param reply reply 
+   */
+  const buildRow = (reply: Reply, fields: MetaformField[]) => {
+    const row : { [key: string]: string | number } = {};
+    
+    row.replyId = reply.id!;
```
Comment (anttileppa) [inline src/components/screens/forms/form-replies-screen.tsx:48]: Each DataGrid row should have id -property

File: src/components/screens/forms/form-replies-screen.tsx [hunk 8]
```diff
@@ -0,0 +1,278 @@
...
+
+    fields.forEach(field => {
```
Comment (anttileppa) [inline src/components/screens/forms/form-replies-screen.tsx:50]: Couldn't this be replaced by map?

**Review by anttileppa — CHANGES_REQUESTED** [review]

File: src/components/screens/forms/form-replies-screen.tsx (hunk 4 above)
Comment (EVP1989) [reply src/components/screens/forms/form-replies-screen.tsx:40]: formId is checked for null in setup method

File: src/components/screens/forms/form-replies-screen.tsx (hunk 7 above)
Comment (EVP1989) [reply src/components/screens/forms/form-replies-screen.tsx:50]: Added uuid as row.id

File: src/components/screens/forms/form-replies-screen.tsx (hunk 8 above)
Comment (EVP1989) [reply src/components/screens/forms/form-replies-screen.tsx:54]: It doesn't return anything so map would trigger sonar cloud error

File: src/components/screens/forms/form-replies-screen.tsx (hunk 7 above)
Comment (anttileppa) [reply src/components/screens/forms/form-replies-screen.tsx:50]: That's not stable, just rename reployId -> id

**Review by anttileppa — CHANGES_REQUESTED** [review]

**Review by anttileppa — APPROVED** [review]

**Pull request merged**